pkg/server/registry/apigroups/acorn/events/strategy.go

```diff
@@ -182,6 +221,76 @@ func stripQuery(opts storage.ListOptions) (q query, stripped storage.ListOptions
 	return
 }
 
+// parseTimeBound parses a time bound from a string.
+//
+// It attempts to parse raw as one of the following formats, in order, returning the result of the first successful parse:
+// 1. Go duration; e.g. "5m"
+//   - time is calculated relative to now
+//   - if since is true, then the duration is subtracted from now, otherwise it is added
+//
+// 2. RFC3339; e.g. "2006-01-02T15:04:05Z07:00"
+// 3. RFC3339Micro; e.g. "2006-01-02T15:04:05.999999Z07:00"
+// 4. Unix timestamp; e.g. "1136239445"
+func parseTimeBound(raw string, now internalv1.MicroTime, since bool) (*internalv1.MicroTime, error) {
+	// Try to parse raw as a duration string
+	var errs []error
+	duration, err := time.ParseDuration(raw)
+	if err == nil {
+		if since {
+			duration *= -1
+		}
+
+		return z.P(internalv1.NewMicroTime(now.Add(duration))), nil
+	}
+	errs = append(errs, fmt.Errorf("%s is not a valid duration: %w", raw, err))
+
+	// Try to parse raw as a time string
+	t, err := parseTime(raw)
+	if err == nil {
+		return t, nil
+	}
+	errs = append(errs, fmt.Errorf("%s is not a valid time: %w", raw, err))
+
+	// Try to parse raw as a unix timestamp
+	unix, err := parseUnix(raw)
+	if err == nil {
+		return unix, nil
+	}
+	errs = append(errs, fmt.Errorf("%s is not a valid unix timestamp: %w", raw, err))
+
+	return nil, errors.Join(errs...)
+}
+
+var (
+	supportedLayouts = []string{
+		time.RFC3339,
+		metav1.RFC3339Micro,
+	}
+)
+
+func parseTime(raw string) (*internalv1.MicroTime, error) {
+	var errs []error
+	for _, layout := range supportedLayouts {
+		since, err := time.Parse(layout, raw)
+		if err == nil {
+			return z.P(internalv1.NewMicroTime(since)), nil
```

Comment:
nit: this is super minor but imo `since` is a confusing variable name here

```suggestion
		t, err := time.Parse(layout, raw)
		if err == nil {
			return z.P(internalv1.NewMicroTime(t)), nil
```